### ISSUE TITLE
Atom 0.192 Deprecations

### ISSUE THREAD

**Issue by vindicatorr** [OmniSharp/omnisharp-atom#101]
<hr>
Use `activationCommands` instead of `activationEvents` in your package.json\nCommands should be grouped by selector as follows:\n```json\n  \"activationCommands\": {\n    \"atom-workspace\": [\"foo:bar\", \"foo:baz\"],\n    \"atom-text-editor\": [\"foo:quux\"]\n  }\n```
./package.json:6
  \-activationEvents
  \+activationCommands
<hr>
<hr>
"ContextMenuManager::add has changed to take a single object as its\nargument. Please see\nhttps://atom.io/docs/api/latest/ContextMenuManager for more info."
 ./menus/atom-sharper.cson:2
  'context-menu': {
   '.overlayer': [
     { 'label': 'Enable atom-sharp', 'command': 'omnisharp-atom:toggle' }
     { 'label': 'Find Usages', 'command': 'omnisharp-atom:find-usages' }
     { 'label': 'Rename', 'command': 'omnisharp-atom:rename' }
    ]
   }
I don't think this is right though. I don't know the exact purpose of it, but am thinking it is a right-mouse click on the omni flame at the bottom which currently does nothing.
<hr>
<hr>
"Are you trying to listen for the '" + types + "' Atom command with `jQuery::on`?\n`jQuery::trigger` can no longer be used to listen for Atom commands. Please\nuse `atom.commands.add` instead. See the docs at\nhttps://atom.io/docs/api/latest/CommandRegistry#instance-add for details."
 ./node_modules/jquery/dist/jquery.js:4856:17
  \- jquery.events.add
  \+ atom.commands.add
<hr>
<hr>
"Use version ^1.0.0 of the status-bar Service API."
 ./package.json:29
  \- ^0.58.0
  \+ ^1.0.0
<hr>

The deprecations are cleared at this point.
I click on the flame at the bottom and the omni status box comes up. I click inside the "Omnisharp server is turned off" (not the button), and 2 more deprecations pop up:

<hr>
atom.workspaceView is no longer available.
In most cases you will not need the view. See the Workspace docs for
alternatives: https://atom.io/docs/api/latest/Workspace.
If you do need the view, please use `atom.views.getView(atom.workspace)`,
which returns an HTMLElement.
```
Atom.Object.defineProperty.get (/usr/local/share/atom/resources/app/src/atom.js:55:13)
DockView.module.exports.DockView.resizeStarted (/home/user/.atom/packages/omnisharp-atom/lib/omnisharp-atom/views/dock-view.coffee:88:28)
```
  \- atom.workspaceView.statusBar.height()
  \+ atom.views.getView(atom.workspace).statusBar.height()
  This "suggested" change fails too though with "height" being undefined.
   I need to sleep now.
<hr>
<hr>
The atom.workspaceView.statusBar global is deprecated. The global was
previously being assigned by the status-bar package, but Atom packages
should never assign globals.

In the future, this problem will be solved by an inter-package communication
API available on `atom.services`. For now, you can get a reference to the
`status-bar` element via `document.querySelector('status-bar')`.
```
WorkspaceView.Object.defineProperty.get (/usr/local/share/atom/resources/app/node_modules/status-bar/lib/main.js:68:20)
DockView.module.exports.DockView.resizeStarted (/home/user/.atom/packages/omnisharp-atom/lib/omnisharp-atom/views/dock-view.coffee:88:42)
```
  \- atom.workspaceView.statusBar.height()
  \+ atom.views.getView(atom.workspace).statusBar.height()
//  \+ document.querySelector('status-bar').height()
This hasn't come up with any error (yet).
<hr>

**Comment by willl** [comment]
Hi @vindicatorr, thanks for reporting those. Some of the issues that you've picked up have already been resolved but a new release hasn't been made in a little while.

Eg: 
* PR #93 addresses the statusBar issues.
* PR #63 addresses the workspaceView issue.

The easy wins if anyone wants to send in a PR is to change the two values in package.json:
* Change activationEvents to be activationCommands
* Change consumedServicse -> status-bar from ^0.58.0 to ^1.0.0.

The others may take a bit longer to address.

**Comment by willl** [comment]
From what I can tell, the only remaining issues listed by @vindicatorr after #102 got merged into master is the one relating to jQuery::on

**Comment by ctolkien** [comment]
jQuery::on seems to be in space-pen views, not omnisharp-atom?

**Comment by david-driscoll** [comment]
might be us.

in dock-view.ts
```csharp
this.on('core:cancel core:close', () => this.hideView());
```
